### ISSUE TITLE
Senior withdrawals may stall under extreme loss when junior balance is fee-enriched beyond gross pool principal

### ISSUE THREAD

**Issue by v1ktorrr0x** [dcccrypto/percolator-stake#157]
## Code

`effective_junior_balance` computes the senior baseline using principal-only `gross_pool`:

```rust
// state.rs — effective_junior_balance
let gross_pool = self.total_deposited.saturating_sub(self.total_withdrawn);
let gross_senior = gross_pool.saturating_sub(jb); // jb = junior_balance(), fee-inclusive
```

`junior_balance()` grows with fee distributions via `accrue_fees_inner`:

```rust
// accrue_fees_inner — tranche fee distribution
pool.set_junior_balance(
    pool.junior_balance().checked_add(junior_fee)...
);
```

When accumulated junior fees cause `junior_balance > gross_pool`, `gross_senior` saturates to zero. `distribute_loss` then applies the full `net_loss` to junior alone with no spillover. If `net_loss > junior_balance` (extreme loss), `effective_junior_balance()` returns zero but `total_pool_value()` can still return a positive value (residual fees). `senior_balance()` = `total_pool_value() - 0` = positive, yet the vault physically cannot cover it.

## Impact

Senior withdrawals that compute a positive `withdrawal_amount` via `calc_senior_collateral_for_withdraw` will fail at the `spl_token::transfer` CPI due to insufficient vault balance. No funds are lost — the transfer is rejected — but senior withdrawals are stuck until the pool is made whole (e.g. via `ReturnInsurance` or new deposits). This is a liveness failure under extreme over-loss conditions, not a theft vector.

Conditions required:
1. Pool has tranches enabled with significant fee history flowing to junior
2. `net_loss > junior_balance` (loss exceeds the fee-enriched junior sub-pool)
3. Residual fees keep `total_pool_value()` positive

## Fix

In `effective_junior_balance`, use `total_pool_value()` as the loss baseline rather than `gross_pool`, so the senior backstop correctly accounts for fee contributions:

```rust
let pool_value = self.total_pool_value()?;
let gross_senior = pool_value.saturating_sub(jb);
```

Alternatively, cap `senior_balance()` to the actual vault balance to prevent the accounting from exceeding what is physically withdrawable.

**Comment by dcccrypto** [comment]
On verification this does **not** appear to be a reachable bug. The gross_senior saturation in `effective_junior_balance` only changes the loss split when `net_loss > jb` (junior can't cover, senior must absorb the excess). But a fee-enriched junior has `jb > principal`, and `net_loss = total_flushed − total_returned ≤ total_flushed ≤ total_deposited ≤ principal < jb` — so junior (first-loss) **always** covers the entire loss, and the saturation never affects the outcome (senior stays exactly at its principal). I couldn't construct any input where senior_balance actually stalls/underflows. Reverted the speculative gross_pool change (didn't want to ship an unverified change to fund-distribution code). **If you have a concrete repro** (specific deposited/withdrawn/fees/flushed where senior_balance returns None or mis-distributes), please add it and I'll fix immediately.

**Comment by v1ktorrr0x** [comment]
Confirmed — the invariant holds. `FlushToInsurance` is bounded by `available = total_deposited − total_withdrawn − total_flushed + total_returned`, which equals the actual vault balance. Since `junior_balance` tracks that same allocation (principal + distributed fees), `net_loss ≤ available ≤ junior_balance` at all times. The `gross_senior` saturation path requires `net_loss > junior_balance`, which is unreachable under the current accounting constraints.

No concrete repro exists. Closing as not reachable.
